Empty scenario header causes exception

## Example

```markdown
# 
```

(there is a single space character after the `#`)

## Output

```
// failed to execute phase 'CodeGenerator' due to exception:
// java.lang.StringIndexOutOfBoundsException
// 	(12 internal calls)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:51)
// 	at org.fulib.scenarios.ast.CompilationContext.accept(CompilationContext.java:24)
// 	(3 internal calls)
// 	at org.fulib.webapp.tool.Tools.scenarioc(Tools.java:83)
// 	at org.fulib.webapp.tool.Tools.genCompileRun(Tools.java:161)
// 	at org.fulib.webapp.tool.RunCodeGen.run(RunCodeGen.java:148)
// 	at org.fulib.webapp.tool.RunCodeGen.handle(RunCodeGen.java:50)
// 	(21 internal calls)
// Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 0
// 	(2 internal calls)
// 	at org.fulib.yaml.StrUtil.downFirstChar(StrUtil.java:55)
// 	at org.fulib.yaml.IdMap.generateUniqueKey(IdMap.java:203)
// 	at org.fulib.yaml.IdMap.addToObjIdMap(IdMap.java:188)
// 	at org.fulib.yaml.IdMap.putObject(IdMap.java:115)
// 	at org.fulib.yaml.YamlIdMap.collectObjects(YamlIdMap.java:746)
// 	at org.fulib.yaml.YamlIdMap.encode(YamlIdMap.java:777)
// 	at org.fulib.AbstractGenerator.saveNewClassModel(AbstractGenerator.java:186)
// 	at org.fulib.AbstractGenerator.generate(AbstractGenerator.java:110)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:152)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:31)
// 	at org.fulib.scenarios.ast.ScenarioGroup.accept(ScenarioGroup.java:26)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.lambda$visit$0(CodeGenerator.java:55)
// 	(9 internal calls)
```

--- a/src/main/java/org/fulib/scenarios/visitor/resolve/NameResolver.java
+++ b/src/main/java/org/fulib/scenarios/visitor/resolve/NameResolver.java
@@ -12,6 +12,7 @@ import org.fulib.scenarios.ast.type.PrimitiveType;
 import org.fulib.scenarios.diagnostic.Marker;
 import org.fulib.scenarios.diagnostic.Position;
 import org.fulib.scenarios.parser.Identifiers;
+import org.fulib.util.Validator;
 
 import java.util.*;
 import java.util.function.BiConsumer;
@@ -113,7 +114,13 @@ public enum NameResolver implements CompilationContext.Visitor<Object, Object>, 
    public Object visit(Scenario scenario, Scope par)
    {
       final ClassDecl classDecl = scenario.getFile().getClassDecl();
-      final String methodName = Identifiers.toLowerCamelCase(scenario.getName());
+      String methodName = Identifiers.toLowerCamelCase(scenario.getName());
+
+      if (!Validator.isSimpleName(methodName))
+      {
+         methodName = "_" + methodName;
+      }
+
       final SentenceList body = scenario.getBody();
       final MethodDecl methodDecl = MethodDecl.of(classDecl, methodName, null, PrimitiveType.VOID, body);
       final Position position = scenario.getPosition();

